dbms/src/TestUtils/InterpreterTestUtils.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/FmtUtils.h>
+#include <Common/StringUtils/StringUtils.h>
+#include <Debug/MockStorage.h>
+#include <Flash/executeQuery.h>
+#include <Poco/File.h>
+#include <Poco/FileStream.h>
+#include <TestUtils/InterpreterTestUtils.h>
+#include <TestUtils/executorSerializer.h>
+
+#include <string>
+
+namespace DB::tests
+{
+namespace
+{
+std::vector<std::string> stringSplit(const String & str, char delim)
+{
+    std::stringstream ss(str);
+    std::string item;
+    std::vector<std::string> elems;
+    while (std::getline(ss, item, delim))
+    {
+        if (!item.empty())
+            elems.push_back(item);
+    }
+    return elems;
+}
+
+String getOutputPath()
+{
+    const auto & test_info = testing::UnitTest::GetInstance()->current_test_info();
+    assert(test_info);
+    String file_name = test_info->file();
+    auto out_path = file_name.replace(file_name.find(".cpp"), 4, ".out");
+    Poco::File file(out_path);
+    if (!file.exists())
+        file.createFile();
+    return out_path;
+}
+} // namespace
+
+void InterpreterTestUtils::initExpectResults()
+{
+    if (just_record)
+    {
+        case_expect_results.clear();
+        return;
+    }
+
+    auto out_path = getOutputPath();
```

Comment:
What will happen when "*.out" files are deleted and user forget to call set_record?

Reply:
It will print ""can not find expect result\n"<< test_info_msg();"